internal/browser/executor.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package browser
+
+import (
+	"context"
+	"math"
+	"testing"
+
+	"github.com/chromedp/cdproto/emulation"
+	"github.com/chromedp/cdproto/page"
+	"github.com/chromedp/chromedp"
+)
+
+// defaultOptions are the default Chrome options.
+var defaultOptions = [...]chromedp.ExecAllocatorOption{
+	chromedp.NoFirstRun,
+	chromedp.NoDefaultBrowserCheck,
+
+	// After Puppeteer's default behavior.
+	chromedp.Flag("disable-background-networking", true),
+	chromedp.Flag("enable-features", "NetworkService,NetworkServiceInProcess"),
+	chromedp.Flag("disable-background-timer-throttling", true),
+	chromedp.Flag("disable-backgrounding-occluded-windows", true),
+	chromedp.Flag("disable-breakpad", true),
+	chromedp.Flag("disable-client-side-phishing-detection", true),
+	chromedp.Flag("disable-default-apps", true),
+	chromedp.Flag("disable-dev-shm-usage", true),
+	chromedp.Flag("disable-extensions", true),
+	chromedp.Flag("disable-features", "site-per-process,TranslateUI,BlinkGenPropertyTrees"),
+	chromedp.Flag("disable-hang-monitor", true),
+	chromedp.Flag("disable-ipc-flooding-protection", true),
+	chromedp.Flag("disable-popup-blocking", true),
+	chromedp.Flag("disable-prompt-on-repost", true),
+	chromedp.Flag("disable-renderer-backgrounding", true),
+	chromedp.Flag("disable-sync", true),
+	chromedp.Flag("force-color-profile", "srgb"),
+	chromedp.Flag("metrics-recording-only", true),
+	chromedp.Flag("safebrowsing-disable-auto-update", true),
+	chromedp.Flag("enable-automation", true),
+	chromedp.Flag("password-store", "basic"),
+	chromedp.Flag("use-mock-keychain", true),
+}
+
+// NewContext creates a new browser instance with the given options. Set
+// `headless` to false to watch the actual browser interaction. All future calls
+// to `Run` must use the context returned by this function!
+//
+// If this function returns successfully, a browser is running and ready to be
+// used. It's recommended that you wrap the returned function in a timeout.
+func NewContext(tb testing.TB, headless bool) context.Context {
```

Comment:
This is a personal nit.

I hate functions that take random bools. It prevents readers of the calling code from understanding what is happening. I prefer to either hide this in `NewHeadlessContext` and `NewHeadedContext` or a type for the bool.

Reply:
Done